packages/remix/src/utils/instrumentServer.ts

```diff
@@ -321,16 +322,18 @@ function wrapRequestHandler(origRequestHandler: RequestHandler, build: ServerBui
   const routes = createRoutes(build.routes);
   const pkg = loadModule<ReactRouterDomPkg>('react-router-dom');
   return async function (this: unknown, request: Request, loadContext?: unknown): Promise<Response> {
-    const url = new URL(request.url);
+    const local = domain.create();
+    return local.bind(async () => {
+      const url = new URL(request.url);
+      const transaction = startRequestHandlerTransaction(url, request.method, routes, pkg);
 
-    const transaction = startRequestHandlerTransaction(url, request.method, routes, pkg);
+      const res = (await origRequestHandler.call(this, request, loadContext)) as Response;
 
-    const res = (await origRequestHandler.call(this, request, loadContext)) as Response;
+      transaction?.setHttpStatus(res.status);
```

Comment:
Under what circumstances would `transaction` be undefined? (Applies to express as well.)